psydac/core/bsplines.py

```diff
@@ -454,6 +465,7 @@ def greville(knots, degree, periodic, out=None):
         Abscissas of all Greville points.
 
     """
+    knots = np.asarray(knots)
```

Comment:
As above

Reply:
Done in https://github.com/pyccel/psydac/commit/54fa1fb4c2afba79c77392308a193bad59161fee